spec/acceptance/resolved_spec.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require 'spec_helper_acceptance'
+
+describe 'systemd' do
+  context 'configure systemd resolved' do
+    it 'works idempotently with no errors' do
+      pp = <<-PUPPET
+      class{'systemd':
+        manage_resolved    => true,
+        manage_resolv_conf => false,
+      }
+      PUPPET
+      apply_manifest(pp, catch_failures: true)
+      # RedHat 9 requires an extra run
+      apply_manifest(pp, catch_failures: true)
```

Comment:
can you wrap that in a condition so it only runs twice on RHEL9? Also why are two runs required? that shouldn't be the case at all?

Reply:
Perhaps rephrase it to "extra run to populate $facts['internal_services']"?